src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -588,7 +588,38 @@ public OptionalLong indexOf(Predicate<? super T> predicate) {
             return a == null ? null : StreamEx.of(Arrays.spliterator(a));
         });
     }
-    
+
+    /**
+     * Returns a stream consisting of the results of replacing each element of
+     * this stream with the contents of the optional value produced by applying
+     * the provided mapping function to each element.
+     *
+     * <p>
+     * If the mapping function returns {@link Optional#empty()}, the original
+     * value will be removed from the resulting stream.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">intermediate
+     * operation</a>.
+     *
+     * <p>
+     * The {@code flatOption()} operation has the effect of applying a
+     * one-to-zero-or-one transformation to the elements of the stream, and then
+     * flattening the resulting elements into a new stream.
+     *
+     * @param <R> The element type of the new stream
+     * @param mapper a <a
+     *        href="package-summary.html#NonInterference">non-interfering </a>,
+     *        <a href="package-summary.html#Statelessness">stateless</a>
+     *        function to apply to each element which produces an
+     *        array of new values
+     * @return the new stream
+     * @since 0.6.8
+     */
+    public <R> StreamEx<R> flatOption(Function<? super T, ? extends Optional<? extends R>> mapper) {
```

Comment:
I don't like `flatOption`, because there's no "option" in Java. Why not `flatOptional`? Or probably something different like `mapPartial` (thinking of function returning `Optional` as of partial function)?